crates/swc_isograph_plugin/src/lib.rs

```diff
@@ -0,0 +1,458 @@
+            .expect("Expected path to be diffable");
+
+        let mut file_to_artifact = PathBuf::from(format!(
+            // TODO a bug in the babel transform: https://github.com/isographlabs/isograph/issues/496
+            "{}/{}/{}/{}.ts",
+            file_to_artifact_dir.display(),
+            self.field_type,
+            self.field_name,
+            self.artifact_type
+        ));
+
+        if cfg!(target_os = "windows") {
+            file_to_artifact =
+                PathBuf::from(format!("{}", file_to_artifact.display()).replace("\\", "/"));
+        }
+
+        if file_to_artifact.starts_with(ISOGRAPH_FOLDER) {
+            file_to_artifact = PathBuf::from(format!("./{}", file_to_artifact.display()));
+        }
+
+        Ok(file_to_artifact)
+    }
+}
+
+#[derive(Debug, Clone)]
+struct IsoLiteralCompilerVisitor<'a> {
+    root_dir: &'a Path,
+    config: &'a IsographProjectConfig,
+    filepath: &'a Path,
+    imports: Vec<IsographImport>,
+    unresolved_mark: Option<Mark>,
+}
+
+#[swc_trace]
+impl IsoLiteralCompilerVisitor<'_> {
+    fn parse_iso_call_arg_into_type(
+        &self,
+        expr_or_spread: &ExprOrSpread,
+    ) -> Result<ValidIsographTemplateLiteral, IsographTransformError> {
+        if let Expr::Tpl(Tpl { quasis, .. }) = &*expr_or_spread.expr {
+            let first = if let Some((first, [])) = quasis.split_first() {
+                first
+            } else {
+                return Err(IsographTransformError::SubstitutionsNotAllowedInIsoFragments);
+            };
+
+            return OPERATION_REGEX
+                .captures_iter(first.raw.trim())
+                .next()
+                .map(|capture_group| {
+                    debug!("capture_group {:?}", capture_group);
+                    ValidIsographTemplateLiteral {
+                        artifact_type: ArtifactType::from(capture_group[1].to_string()),
+                        field_type: capture_group[2].to_string(),
+                        field_name: capture_group[3].to_string(),
+                    }
+                })
+                .ok_or(IsographTransformError::InvalidIsoKeyword);
+        }
+        Err(IsographTransformError::OnlyAllowedTemplateLiteral)
+    }
+
+    fn valid_isograph_template_literal(
+        &mut self,
+        entrypoint: &ValidIsographTemplateLiteral,
+    ) -> Expr {
+        let file_to_artifact = entrypoint
+            .path_for_artifact(self.filepath, self.config, self.root_dir)
+            .expect("Failed to get path for artifact.");
+
+        match self.config.options.module {
+            ConfigFileJavascriptModule::CommonJs => entrypoint.build_require_expr_from_path(
+                &file_to_artifact.display().to_string(),
+                self.unresolved_mark,
+            ),
+            ConfigFileJavascriptModule::EsModule => {
+                let ident_name = format!("_{}", entrypoint.field_name);
```

Comment:
do we have an easy way to generating an unused import name? if not this is fine, lets just leave a 
```
TODO ensure `ident_name` is unique
```